src/screencast/screencast.c

```diff
@@ -585,12 +585,73 @@ static int method_screencast_start(sd_bus_message *msg, void *data,
 	if (ret < 0) {
 		return ret;
 	}
-	ret = sd_bus_message_append(reply, "{sv}",
-		"streams", "a(ua{sv})", 1,
-		cast->node_id, 3,
-		"position", "(ii)", 0, 0,
-		"size", "(ii)", cast->screencopy_frame_info[WL_SHM].width, cast->screencopy_frame_info[WL_SHM].height,
-		"source_type", "u", MONITOR);
+	ret = sd_bus_message_open_container(reply, 'e', "sv");
+	if (ret < 0) {
+		return ret;
+	}
+	ret = sd_bus_message_append(reply, "s", "streams");
+	if (ret < 0) {
+		return ret;
+	}
+	ret = sd_bus_message_open_container(reply, 'v', "a(ua{sv})");
+	if (ret < 0) {
+		return ret;
+	}
+	ret = sd_bus_message_open_container(reply, 'a', "(ua{sv})");
+	if (ret < 0) {
+		return ret;
+	}
+	ret = sd_bus_message_open_container(reply, 'r', "ua{sv}");
+	if (ret < 0) {
+		return ret;
+	}
+	ret = sd_bus_message_append(reply, "u", cast->node_id);
+	if (ret < 0) {
+		return ret;
+	}
+	ret = sd_bus_message_open_container(reply, 'a', "{sv}");
+	if (ret < 0) {
+		return ret;
+	}
+	if (cast->target->output->xdg_output) {
+		ret = sd_bus_message_append(reply, "{sv}",
+			"position", "(ii)", cast->target->output->x, cast->target->output->y);
+		if (ret < 0) {
+			return ret;
+		}
+		ret = sd_bus_message_append(reply, "{sv}",
+			"size", "(ii)", cast->target->output->width, cast->target->output->height);
+		if (ret < 0) {
+			return ret;
+		}
+	} else {
```

Comment:
We should probably still send "position" when xdg-output is unavailable? The D-Bus API docs suggest this is always set for MONITOR?

Reply:
Oh, you replied above. And indeed, it's marked optional here. I was looking at the backend API docs where the "optional" word is missing.

Yeah, I prefer to leave it out if we don't have the info, rather than send zero.

Can we also skip sending the size if we don't have xdg-output?

Reply:
> Can we also skip sending the size if we don't have xdg-output?

Sure, if you think that's best.

For the record, `xdg-desktop-portal` does try to pull out [the size](https://github.com/flatpak/xdg-desktop-portal/blob/main/src/screen-cast.c#L721), which then appears to [be used](https://github.com/flatpak/xdg-desktop-portal/blob/main/src/screen-cast.c#L693C18-L693C18) by the [remote desktop](https://github.com/flatpak/xdg-desktop-portal/blob/main/src/remote-desktop.c#L821) code to determine if pointer/touch events are in bounds. So if #263 or similar is merged it would have ramifications on compositors that don't support xdg-output.

Reply:
If the docs say it's optional but xdp itself requires it, I'd argue that it's an xdg bug :P